### ISSUE TITLE
Vocabularies that are restricted from loading with platforms.txt keep throwing the error when you refresh-all

### ISSUE THREAD

**Issue by erg** [factor/factor#179]
On Mac:

```
USE: windows
-- error message here
refresh-all
-- error message here, etc until you << "windows" forget-vocab >>
```

**Comment by mrjbq7** [comment]
This fixes it, but I want to think about whether this ``check-vocab-hook`` should be somewhere else like ``create-vocab``...

```diff
diff --git a/core/vocabs/loader/loader.factor b/core/vocabs/loader/loader.factor
index b1d6e31359..e0b8b12aea 100644
--- a/core/vocabs/loader/loader.factor
+++ b/core/vocabs/loader/loader.factor
@@ -159,7 +159,9 @@ M: vocab (require)
 M: vocab-link (require)
     vocab-name (require) ;
 
-M: string (require) create-vocab (require) ;
+M: string (require)
+    dup check-vocab-hook get call( vocab -- )
+    create-vocab (require) ;
 
 PRIVATE>
 
```

**Comment by mrjbq7** [comment]
Fixed by 38eab40848539422e3ea764bed331d21a0ce8050.
